tests/test_java.py

```diff
@@ -3,7 +3,7 @@ def test_java_is_runnable_via_java_home_env_var(host):
 
 
 def test_java_major_version(host):
-    assert 'openjdk version "11.' in host.run('$JAVA_HOME/bin/java -version').stderr
+    assert 'openjdk version "11' in host.run('$JAVA_HOME/bin/java -version').stderr
```

Comment:
This is really brittle. Could we take a different approach please and allow this to match `11((\.0)*\.[1-9][0-9]*)*`? And then we will be good until JDK 12.

Reply:
Not sure I follow, I have done some tests and this will work until JDK12 because it tests the first part of the string.

Reply:
I gave a more precise regular expression that will only match JDK builds for JDK 11 that conform to the official version string specification. I don’t like how we had to change this for 10.0.2 to 11 and then got hung up by the fact that the first release in the 11 series does not have a dot after the major version number. With the change I propose, we get a strong assertion, and only have to make a minor update when we go from 11 to 12.

Reply:
got it